exporters/metric/cortex/go.mod

```diff
@@ -7,7 +7,7 @@ require (
 	github.com/golang/snappy v0.0.2
 	github.com/google/go-cmp v0.5.2
 	github.com/grpc-ecosystem/grpc-gateway v1.14.6 // indirect
-	github.com/prometheus/prometheus v2.5.0+incompatible
+	github.com/prometheus/prometheus v0.0.0-20200911110723-e83ef207b6c2
```

Comment:
Can we have this refer to a tagged release? It looks like 2.21.0 is the latest release, would that work?

Reply:
There is no tag in Prometheus project. Refer to cortex and thanos
![image](https://user-images.githubusercontent.com/9583245/94297070-7c12b700-ff96-11ea-8823-32b7f7fee268.png)
cortex
![image](https://user-images.githubusercontent.com/9583245/94297174-a9f7fb80-ff96-11ea-80d2-02afdd4fa261.png)
thanos

Reply:
![image](https://user-images.githubusercontent.com/9583245/94297800-aca72080-ff97-11ea-9eeb-f7737bee2add.png)
![image](https://user-images.githubusercontent.com/9583245/94297830-baf53c80-ff97-11ea-84cf-6f850afa1fd6.png)
This is 2.21.0

Reply:
EasyCLA done

Reply:
@liguozhong thanks for explaining your thought process. In this project we default to using tagged releases as dependencies when possible. This helps to ensure the code used is stable as well as communicate this stability in a discoverable manner in our dependency manifests. Please go ahead and update to the tagged release version of the prometheus package that is the equivalent of the hash currently referenced.

Reply:
I tried to use a tagged release for `prometheus/prometheus` in another context, but there's a problem. You get an error from the Go toolchain:

```
server response: not found: github.com/prometheus/prometheus@v2.21.0+incompatible: invalid version: +incompatible suffix not allowed: module contains a go.mod file, so semantic import versioning is required
```
I haven't figured out what to do about this, it suggests some degree of restructuring in the Prometheus project, and my guess is the other projects shown above have already raised issues about this.

Reply:
Also note: the reason we're on v2.5.0+incompatible is that that's the last version before a `go.mod` file appeared.

Reply:
I see. It looks like with semantic import versioning [that the module system enforces](https://golang.org/cmd/go/#hdr-Module_compatibility_and_semantic_versioning) `v2` versions of a package are assumed to be in a separate `v2` directory. The Prometheus community made their v2 release prior to module system and semantic import versioning were a thing and do not follow this convention.